packages/@webex/plugin-meetings/src/reachability/clusterReachability.ts

```diff
@@ -0,0 +1,324 @@
+      url.searchParams.append('transport', 'tcp');
+
+      return {
+        username: 'webexturnreachuser',
+        credential: 'webexturnreachpwd',
+        urls: [url.toString()],
+      };
+    });
+
+    return {
+      iceServers: [...udpIceServers, ...tcpIceServers],
+      iceCandidatePoolSize: 0,
+      iceTransportPolicy: 'all',
+    };
+  }
+
+  /**
+   * Creates an RTCPeerConnection
+   * @param {ClusterNode} clusterInfo information about the media cluster
+   * @returns {RTCPeerConnection} peerConnection
+   */
+  private createPeerConnection(clusterInfo: ClusterNode) {
+    try {
+      const config = this.buildPeerConnectionConfig(clusterInfo);
+
+      const peerConnection = new RTCPeerConnection(config);
+
+      return peerConnection;
+    } catch (peerConnectionError) {
+      LoggerProxy.logger.log(
+        `Reachability:index#createPeerConnection --> Error creating peerConnection: ${peerConnectionError}`
+      );
+
+      return undefined;
+    }
+  }
+
+  /**
+   * @returns {ClusterReachabilityResult} reachability result for this cluster
+   */
+  getResult() {
+    return this.result;
+  }
+
+  /**
+   * Closes the peerConnection
+   *
+   * @returns {void}
+   */
+  private closePeerConnection() {
+    if (this.pc) {
+      this.pc.onicecandidate = null;
+      this.pc.onicegatheringstatechange = null;
+      this.pc.close();
+    }
+  }
+
+  /**
+   * Resolves the defer, indicating that reachability checks for this cluster are completed
+   *
+   * @returns {void}
+   */
+  private finishReachabilityCheck() {
+    this.defer.resolve();
+  }
+
+  /**
+   * Adds public IP (client media IPs)
+   * @param {string} protocol
+   * @param {string} publicIP
+   * @returns {void}
+   */
+  private addPublicIP(protocol: 'udp' | 'tcp', publicIP?: string | null) {
+    const result = this.result[protocol];
+
+    if (publicIP) {
+      if (result.clientMediaIPs) {
+        if (!result.clientMediaIPs.includes(publicIP)) {
+          result.clientMediaIPs.push(publicIP);
+        }
+      } else {
+        result.clientMediaIPs = [publicIP];
+      }
+    }
+  }
+
+  /**
+   * Registers a listener for the iceGatheringStateChange event
+   *
+   * @returns {void}
+   */
+  private registerIceGatheringStateChangeListener() {
+    this.pc.onicegatheringstatechange = () => {
+      const {COMPLETE} = ICE_GATHERING_STATE;
+
+      if (this.pc.iceConnectionState === COMPLETE) {
+        this.closePeerConnection();
+        this.finishReachabilityCheck();
+      }
+    };
+  }
+
+  /**
+   * Checks if we have the results for all the protocols (UDP and TCP)
+   *
+   * @returns {boolean} true if we have all results, false otherwise
+   */
+  private haveWeGotAllResults(): boolean {
+    return ['udp', 'tcp'].every(
+      (protocol) =>
+        this.result[protocol].result === 'reachable' ||
+        this.result[protocol].result === 'not tested'
+    );
+  }
+
+  /**
+   * Stores the latency in the result
+   *
```

Comment:
suggestion: adding to the jsdoc that this is what marks the result as 'reachable' might be useful as well

Reply:
done